CO-48 Create Grafana dashboard (#63)
containeroo/cloudflare-operator: docs/content/monitoring.md

# Monitoring with Prometheus

> ## prerequisites

Reviewer: ```suggestion
## Prerequisites
```

- [Prometheus](https://github.com/prometheus-community/helm-charts/tree/main/charts/prometheus) - collects metrics from the cloudflare-operator controllers and Kubernetes API
- [Grafana](https://github.com/grafana/helm-charts/tree/main/charts/grafana) dashboards - displays cloudflare-operator stats
- [kube-state-metrics](https://github.com/prometheus-community/helm-charts/tree/main/charts/kube-state-metrics) - generates metrics about the state of the Kubernetes objects

> The easiest way to deploy all necessary applications is to use [kube-prometheus-stack](https://github.com/prometheus-community/helm-charts/tree/main/charts/kube-prometheus-stack).

Reviewer: ```suggestion
The easiest way to deploy all necessary components is to use [kube-prometheus-stack](https://github.com/prometheus-community/helm-charts/tree/main/charts/kube-prometheus-stack).
```

> ## Install cloudflare-operator Grafana dashboard

Reviewer: ```suggestion
## Install cloudflare-operator Grafana Dashboard
```

> Note that the cloudflare-operator expose the `/metrics` endpoint on port `8080`. When using Prometheus Operator you need a `PodMonitor` object to configure scraping for the controllers.

Reviewer: ```suggestion
Note that the cloudflare-operator exposes the `/metrics` endpoint on port `8080`. When using Prometheus Operator you need a `PodMonitor` object to configure scraping for the controller.
```

> Apply the `config/manifests/prometheus/monitor.yaml` containing the `PodMonitor` and create a configmap with the cloudflare-operator dashboard:

Reviewer: ```suggestion
Apply the `config/manifests/prometheus/monitor.yaml` containing the `PodMonitor` and create a ConfigMap with the cloudflare-operator dashboard:
```

```bash
# Create a podmonitor
kubectl apply -f https://raw.githubusercontent.com/containeroo/cloudflare-operator/master/config/manifests/prometheus/monitor.yaml

# Download Grafana dashboard
wget https://raw.githubusercontent.com/containeroo/cloudflare-operator/master/config/manifests/grafana/dashboards/overview.json -O /tmp/grafana-dashboard-cloudflare-operator.json

# Create the configmap
kubectl create configmap grafana-dashboard-cloudflare-operator --from-file=/tmp/grafana-dashboard-cloudflare-operator.json

# Add label so Grafana can fetch dashboard
kubectl label configmap grafana-dashboard-cloudflare-operator grafana_dashboard="1"
```

## Metrics

> For each `cf.containeroo.ch` kind, the controllers expose a gauge metric to track the Ready condition status.

Reviewer: ```suggestion
For each `cf.containeroo.ch` kind, the controller exposes a gauge metric to track the status condition.
```
